src/controllers/PendingTransactionController.ts

```diff
@@ -0,0 +1,52 @@
+
+import { Request, Response } from "express";
+import { sendJSONresponse } from "../common/Utils";
+import { Config } from "../common/Config";
+import { Error } from "mongoose";
+let Promise = require("bluebird");
+
+export class PendingTransactionController {
+
+    public readPendingTransaction(req: Request, res: Response) {
+        const validationErrors: any = PendingTransactionController.validateQueryParameters(req);
+
+        if (validationErrors) {
+            sendJSONresponse(res, 400, {validationErrors});
+            return;
+        }
+
+        const address: string = req.query.address;
+
+        Config.web3.eth.getBlock("pending", true)
+        .then((pendings: any) => {
+            return new Promise.filter(pendings.transactions, function(transaction: any) {
+                return (transaction.from === address || transaction.to == address) ? true : false;
```

Comment:
did you consider having a separate function to filter transactions?

Reply:
I did not. I will consider in the future.